### PR TITLE
fix(components): [cascader] add focus and blur event when set filterable

### LINKED ISSUE #11220: [Component] [cascader] el-cascader
Bug Type: **`Component`**

## Environment

- Vue Version: `3.2.25`
- Element Plus Version: `2.2.19`
- Browser / OS: `Chrome 94`
- Build Tool: `Vite`

### Related Component

- `el-cascader`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cCBsYW5nPVwidHNcIj5cbmltcG9ydCB7IHJlZiwgdmVyc2lvbiBhcyB2dWVWZXJzaW9uIH0gZnJvbSAndnVlJ1xuaW1wb3J0IHsgdmVyc2lvbiBhcyBFcFZlcnNpb24gfSBmcm9tICdlbGVtZW50LXBsdXMnXG5pbXBvcnQgeyBFbGVtZW50UGx1cyB9IGZyb20gJ0BlbGVtZW50LXBsdXMvaWNvbnMtdnVlJ1xuXG5jb25zdCBtc2cgPSByZWYoJ0hlbGxvIFdvcmxkIScpXG5jb25zdCBwcm9wcyA9IHsgbXVsdGlwbGU6IHRydWUgfVxuY29uc3Qgb3B0aW9ucyA9IFtcbiAgICAgICAge1xuICAgICAgICAgIHZhbHVlOiAxLFxuICAgICAgICAgIGxhYmVsOiAnQXNpYScsXG4gICAgICAgICAgY2hpbGRyZW46IFtcbiAgICAgICAgICAgIHtcbiAgICAgICAgICAgICAgdmFsdWU6IDIsXG4gICAgICAgICAgICAgIGxhYmVsOiAnQ2hpbmEnLFxuICAgICAgICAgICAgICBjaGlsZHJlbjogW1xuICAgICAgICAgICAgICAgIHsgdmFsdWU6IDMsIGxhYmVsOiAnQmVpamluZycgfSxcbiAgICAgICAgICAgICAgICB7IHZhbHVlOiA0LCBsYWJlbDogJ1NoYW5naGFpJyB9LFxuICAgICAgICAgICAgICAgIHsgdmFsdWU6IDUsIGxhYmVsOiAnSGFuZ3pob3UnIH0sXG4gICAgICAgICAgICAgIF0sXG4gICAgICAgICAgICB9LFxuICAgICAgICAgICAge1xuICAgICAgICAgICAgICB2YWx1ZTogNixcbiAgICAgICAgICAgICAgbGFiZWw6ICdKYXBhbicsXG4gICAgICAgICAgICAgIGNoaWxkcmVuOiBbXG4gICAgICAgICAgICAgICAgeyB2YWx1ZTogNywgbGFiZWw6ICdUb2t5bycgfSxcbiAgICAgICAgICAgICAgICB7IHZhbHVlOiA4LCBsYWJlbDogJ09zYWthJyB9LFxuICAgICAgICAgICAgICAgIHsgdmFsdWU6IDksIGxhYmVsOiAnS3lvdG8nIH0sXG4gICAgICAgICAgICAgIF0sXG4gICAgICAgICAgICB9LFxuICAgICAgICAgICAge1xuICAgICAgICAgICAgICB2YWx1ZTogMTAsXG4gICAgICAgICAgICAgIGxhYmVsOiAnS29yZWEnLFxuICAgICAgICAgICAgICBjaGlsZHJlbjogW1xuICAgICAgICAgICAgICAgIHsgdmFsdWU6IDExLCBsYWJlbDogJ1Nlb3VsJyB9LFxuICAgICAgICAgICAgICAgIHsgdmFsdWU6IDEyLCBsYWJlbDogJ0J1c2FuJyB9LFxuICAgICAgICAgICAgICAgIHsgdmFsdWU6IDEzLCBsYWJlbDogJ1RhZWd1JyB9LFxuICAgICAgICAgICAgICBdLFxuICAgICAgICAgICAgfSxcbiAgICAgICAgICBdLFxuICAgICAgICB9LFxuICAgICAgXVxuY29uc3QgdGVzdCA9ICgpID0+IHtcbiAgYWxlcnQoJ2ZvY3VzIScpXG59XG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8ZWwtY2FzY2FkZXJcbiAgICAgICAgICAgIDpvcHRpb25zPVwib3B0aW9uc1wiXG4gICAgICAgICAgICA6cHJvcHM9XCJwcm9wc1wiXG4gICAgICAgICAgICBjbGVhcmFibGVcbiAgICAgICAgICAgIGZpbHRlcmFibGVcbiAgICAgICAgICAgIG11bHRpcGxlXG4gICAgICAgICAgICBAZm9jdXM9XCJ0ZXN0XCJcbiAgICAgICAgLz5cbiAgPGVsLWNhc2NhZGVyXG4gICAgICAgICAgICA6b3B0aW9ucz1cIm9wdGlvbnNcIlxuICAgICAgICAgICAgOnByb3BzPVwicHJvcHNcIlxuICAgICAgICAgICAgY2xlYXJhYmxlXG4gICAgICAgICAgICBtdWx0aXBsZVxuICAgICAgICAgICAgQGZvY3VzPVwidGVzdFwiXG4gICAgICAgIC8+XG48L3RlbXBsYXRlPlxuIiwiaW1wb3J0X21hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge31cbn0iLCJfbyI6e319)

### Steps to reproduce

The Event of focus:
First cascader is exception,second cascader is work normal.
The reason is property filterable in multiple cascader.

### What is Expected?

Both of cascader same with second

### What is actually happening?

Strange

### Additional comments

The event focus is not work true in multiple.

### PR DESCRIPTION
**Pull request by btea**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

fix #11220

### CODE REVIEW

**Comment by ryuhangyeong** [comment]
@btea Hello~ Can you elaborate more on the issue? I'm not sure what the problem is.

**Comment by btea** [comment]
If `filterable` and `multiple` are set at the same time, `focus` and `blur` events cannot be triggered.
[before](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZGl2IGNsYXNzPVwibS00XCI+XG4gICAgPHA+Q2hpbGQgb3B0aW9ucyBleHBhbmQgd2hlbiBjbGlja2VkIChkZWZhdWx0KTwvcD5cbiAgICA8ZWwtY2FzY2FkZXIgdi1tb2RlbD1cInZhbHVlXCIgOm9wdGlvbnM9XCJvcHRpb25zXCIgQGZvY3VzPVwiZm9jdXNcIiBmaWx0ZXJhYmxlIDpwcm9wcz1cInByb3BzXCIgLz5cbiAgPC9kaXY+XG48L3RlbXBsYXRlPlxuXG48c2NyaXB0IGxhbmc9XCJ0c1wiIHNldHVwPlxuaW1wb3J0IHsgcmVmIH0gZnJvbSAndnVlJ1xuXG5jb25zdCB2YWx1ZSA9IHJlZihbXSlcblxuY29uc3QgcHJvcHMgPSB7XG4gIG11bHRpcGxlOiB0cnVlXG59XG5cbmNvbnN0IGhhbmRsZUNoYW5nZSA9ICh2YWx1ZSkgPT4ge1xuICBjb25zb2xlLmxvZyh2YWx1ZSlcbn1cblxuY29uc3QgZm9jdXMgPSAoKSA9PiB7XG4gIGNvbnNvbGUubG9nKCdmb2N1cycpXG59XG5cbmNvbnN0IG9wdGlvbnMgPSBbXG4gIHtcbiAgICB2YWx1ZTogJ2d1aWRlJyxcbiAgICBsYWJlbDogJ0d1aWRlJyxcbiAgICBjaGlsZHJlbjogW1xuICAgICAge1xuICAgICAgICB2YWx1ZTogJ2Rpc2NpcGxpbmVzJyxcbiAgICAgICAgbGFiZWw6ICdEaXNjaXBsaW5lcycsXG4gICAgICB9LFxuICAgICAge1xuICAgICAgICB2YWx1ZTogJ25hdmlnYXRpb24nLFxuICAgICAgICBsYWJlbDogJ05hdmlnYXRpb24nLFxuICAgICAgICBjaGlsZHJlbjogW1xuICAgICAgICAgIHtcbiAgICAgICAgICAgIHZhbHVlOiAnc2lkZSBuYXYnLFxuICAgICAgICAgICAgbGFiZWw6ICdTaWRlIE5hdmlnYXRpb24nLFxuICAgICAgICAgIH0sXG4gICAgICAgICAge1xuICAgICAgICAgICAgdmFsdWU6ICd0b3AgbmF2JyxcbiAgICAgICAgICAgIGxhYmVsOiAnVG9wIE5hdmlnYXRpb24nLFxuICAgICAgICAgIH0sXG4gICAgICAgIF0sXG4gICAgICB9LFxuICAgIF0sXG4gIH0sXG4gIHtcbiAgICB2YWx1ZTogJ3Jlc291cmNlJyxcbiAgICBsYWJlbDogJ1Jlc291cmNlJyxcbiAgICBjaGlsZHJlbjogW1xuICAgICAge1xuICAgICAgICB2YWx1ZTogJ2F4dXJlJyxcbiAgICAgICAgbGFiZWw6ICdBeHVyZSBDb21wb25lbnRzJyxcbiAgICAgIH0sXG4gICAgICB7XG4gICAgICAgIHZhbHVlOiAnc2tldGNoJyxcbiAgICAgICAgbGFiZWw6ICdTa2V0Y2ggVGVtcGxhdGVzJyxcbiAgICAgIH0sXG4gICAgICB7XG4gICAgICAgIHZhbHVlOiAnZG9jcycsXG4gICAgICAgIGxhYmVsOiAnRGVzaWduIERvY3VtZW50YXRpb24nLFxuICAgICAgfSxcbiAgICBdLFxuICB9LFxuXVxuPC9zY3JpcHQ+XG4iLCJpbXBvcnRfbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7fVxufSIsIl9vIjp7fX0=)

[after](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZGl2IGNsYXNzPVwibS00XCI+XG4gICAgPHA+Q2hpbGQgb3B0aW9ucyBleHBhbmQgd2hlbiBjbGlja2VkIChkZWZhdWx0KTwvcD5cbiAgICA8ZWwtY2FzY2FkZXIgdi1tb2RlbD1cInZhbHVlXCIgOm9wdGlvbnM9XCJvcHRpb25zXCIgQGZvY3VzPVwiZm9jdXNcIiBmaWx0ZXJhYmxlIDpwcm9wcz1cInByb3BzXCIgLz5cbiAgPC9kaXY+XG48L3RlbXBsYXRlPlxuXG48c2NyaXB0IGxhbmc9XCJ0c1wiIHNldHVwPlxuaW1wb3J0IHsgcmVmIH0gZnJvbSAndnVlJ1xuXG5jb25zdCB2YWx1ZSA9IHJlZihbXSlcblxuY29uc3QgcHJvcHMgPSB7XG4gIG11bHRpcGxlOiB0cnVlXG59XG5cbmNvbnN0IGhhbmRsZUNoYW5nZSA9ICh2YWx1ZSkgPT4ge1xuICBjb25zb2xlLmxvZyh2YWx1ZSlcbn1cblxuY29uc3QgZm9jdXMgPSAoKSA9PiB7XG4gIGNvbnNvbGUubG9nKCdmb2N1cycpXG59XG5cbmNvbnN0IG9wdGlvbnMgPSBbXG4gIHtcbiAgICB2YWx1ZTogJ2d1aWRlJyxcbiAgICBsYWJlbDogJ0d1aWRlJyxcbiAgICBjaGlsZHJlbjogW1xuICAgICAge1xuICAgICAgICB2YWx1ZTogJ2Rpc2NpcGxpbmVzJyxcbiAgICAgICAgbGFiZWw6ICdEaXNjaXBsaW5lcycsXG4gICAgICB9LFxuICAgICAge1xuICAgICAgICB2YWx1ZTogJ25hdmlnYXRpb24nLFxuICAgICAgICBsYWJlbDogJ05hdmlnYXRpb24nLFxuICAgICAgICBjaGlsZHJlbjogW1xuICAgICAgICAgIHtcbiAgICAgICAgICAgIHZhbHVlOiAnc2lkZSBuYXYnLFxuICAgICAgICAgICAgbGFiZWw6ICdTaWRlIE5hdmlnYXRpb24nLFxuICAgICAgICAgIH0sXG4gICAgICAgICAge1xuICAgICAgICAgICAgdmFsdWU6ICd0b3AgbmF2JyxcbiAgICAgICAgICAgIGxhYmVsOiAnVG9wIE5hdmlnYXRpb24nLFxuICAgICAgICAgIH0sXG4gICAgICAgIF0sXG4gICAgICB9LFxuICAgIF0sXG4gIH0sXG4gIHtcbiAgICB2YWx1ZTogJ3Jlc291cmNlJyxcbiAgICBsYWJlbDogJ1Jlc291cmNlJyxcbiAgICBjaGlsZHJlbjogW1xuICAgICAge1xuICAgICAgICB2YWx1ZTogJ2F4dXJlJyxcbiAgICAgICAgbGFiZWw6ICdBeHVyZSBDb21wb25lbnRzJyxcbiAgICAgIH0sXG4gICAgICB7XG4gICAgICAgIHZhbHVlOiAnc2tldGNoJyxcbiAgICAgICAgbGFiZWw6ICdTa2V0Y2ggVGVtcGxhdGVzJyxcbiAgICAgIH0sXG4gICAgICB7XG4gICAgICAgIHZhbHVlOiAnZG9jcycsXG4gICAgICAgIGxhYmVsOiAnRGVzaWduIERvY3VtZW50YXRpb24nLFxuICAgICAgfSxcbiAgICBdLFxuICB9LFxuXVxuPC9zY3JpcHQ+XG4iLCJQbGF5Z3JvdW5kTWFpbi52dWUiOiI8c2NyaXB0IHNldHVwPlxuaW1wb3J0IEFwcCBmcm9tICcuL0FwcC52dWUnXG5pbXBvcnQgeyBzZXR1cEVsZW1lbnRQbHVzIH0gZnJvbSAnLi9lbGVtZW50LXBsdXMuanMnXG5zZXR1cEVsZW1lbnRQbHVzKClcbjwvc2NyaXB0PlxuXG48dGVtcGxhdGU+XG4gIDxBcHAgLz5cbjwvdGVtcGxhdGU+XG4iLCJpbXBvcnRfbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7XG4gICAgXCJlbGVtZW50LXBsdXNcIjogXCJodHRwczovL3ByZXZpZXctMTEyNzMtZWxlbWVudC1wbHVzLnN1cmdlLnNoL2J1bmRsZS9pbmRleC5mdWxsLm1pbi5tanNcIixcbiAgICBcImVsZW1lbnQtcGx1cy9cIjogXCJ1bnN1cHBvcnRlZFwiXG4gIH1cbn0iLCJpbXBvcnQtbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7XG4gICAgXCJ2dWVcIjogXCJodHRwczovL2Zhc3RseS5qc2RlbGl2ci5uZXQvbnBtL0B2dWUvcnVudGltZS1kb21AbGF0ZXN0L2Rpc3QvcnVudGltZS1kb20uZXNtLWJyb3dzZXIuanNcIixcbiAgICBcIkB2dWUvc2hhcmVkXCI6IFwiaHR0cHM6Ly9mYXN0bHkuanNkZWxpdnIubmV0L25wbS9AdnVlL3NoYXJlZEBsYXRlc3QvZGlzdC9zaGFyZWQuZXNtLWJ1bmRsZXIuanNcIixcbiAgICBcImVsZW1lbnQtcGx1c1wiOiBcImh0dHBzOi8vcHJldmlldy0xMTI3My1lbGVtZW50LXBsdXMuc3VyZ2Uuc2gvYnVuZGxlL2luZGV4LmZ1bGwubWluLm1qc1wiLFxuICAgIFwiZWxlbWVudC1wbHVzL1wiOiBcInVuc3VwcG9ydGVkXCIsXG4gICAgXCJAZWxlbWVudC1wbHVzL2ljb25zLXZ1ZVwiOiBcImh0dHBzOi8vZmFzdGx5LmpzZGVsaXZyLm5ldC9ucG0vQGVsZW1lbnQtcGx1cy9pY29ucy12dWVAMi9kaXN0L2luZGV4Lm1pbi5qc1wiXG4gIH0sXG4gIFwic2NvcGVzXCI6IHt9XG59IiwiZWxlbWVudC1wbHVzLmpzIjoiaW1wb3J0IHsgZ2V0Q3VycmVudEluc3RhbmNlIH0gZnJvbSAndnVlJ1xuaW1wb3J0IEVsZW1lbnRQbHVzIGZyb20gJ2VsZW1lbnQtcGx1cydcblxubGV0IGluc3RhbGxlZCA9IGZhbHNlXG5hd2FpdCBsb2FkU3R5bGUoKVxuXG5leHBvcnQgZnVuY3Rpb24gc2V0dXBFbGVtZW50UGx1cygpIHtcbiAgaWYgKGluc3RhbGxlZCkgcmV0dXJuXG4gIGNvbnN0IGluc3RhbmNlID0gZ2V0Q3VycmVudEluc3RhbmNlKClcbiAgaW5zdGFuY2UuYXBwQ29udGV4dC5hcHAudXNlKEVsZW1lbnRQbHVzKVxuICBpbnN0YWxsZWQgPSB0cnVlXG59XG5cbmV4cG9ydCBmdW5jdGlvbiBsb2FkU3R5bGUoKSB7XG4gIHJldHVybiBuZXcgUHJvbWlzZSgocmVzb2x2ZSwgcmVqZWN0KSA9PiB7XG4gICAgY29uc3QgbGluayA9IGRvY3VtZW50LmNyZWF0ZUVsZW1lbnQoJ2xpbmsnKVxuICAgIGxpbmsucmVsID0gJ3N0eWxlc2hlZXQnXG4gICAgbGluay5ocmVmID0gJ2h0dHBzOi8vcHJldmlldy0xMTI3My1lbGVtZW50LXBsdXMuc3VyZ2Uuc2gvYnVuZGxlL2luZGV4LmNzcydcbiAgICBsaW5rLmFkZEV2ZW50TGlzdGVuZXIoJ2xvYWQnLCByZXNvbHZlKVxuICAgIGxpbmsuYWRkRXZlbnRMaXN0ZW5lcignZXJyb3InLCByZWplY3QpXG4gICAgZG9jdW1lbnQuYm9keS5hcHBlbmQobGluaylcbiAgfSlcbn0iLCJfbyI6eyJzaG93SGlkZGVuIjp0cnVlLCJzdHlsZVNvdXJjZSI6Imh0dHBzOi8vcHJldmlldy0xMTI3My1lbGVtZW50LXBsdXMuc3VyZ2Uuc2gvYnVuZGxlL2luZGV4LmNzcyJ9fQ==)

**Comment by ryuhangyeong** [comment]
OK! Thank you for your kind explanation!

**Review by ryuhangyeong — APPROVED** [review]

**Comment by tolking** [comment]
It can also optimize the focus. At present, when you press the Tab key, ElInput is triggered for the first time and input is triggered for the second time. You should use `tabindex` to avoid the first trigger.

**Comment by btea** [comment]
> It can also optimize the focus. At present, when you press the Tab key, ElInput is triggered for the first time and input is triggered for the second time. You should use `tabindex` to avoid the first trigger.

Good catch 👍

**Comment by tolking** [comment]
It seems necessary to add focus class to ElInput

**Comment by btea** [comment]
Do you mean to add `is-focus` here?
https://github.com/element-plus/element-plus/blob/aba0d26537447b936775554e5afe5d9b6d53e33c/packages/components/cascader/src/cascader.vue#L124

**Comment by tolking** [comment]
for ElInput

Currently, there is no style when using the Tab key to focus.

**Comment by tolking** [comment]
The judgment condition here is insufficient. It cannot accurately determine whether the component is in the focus state

[Link](https://github.com/element-plus/element-plus/blob/aba0d26537447b936775554e5afe5d9b6d53e33c/packages/components/cascader/src/cascader.vue#L43)

**Comment by btea** [comment]
I see, indeed as you said, the `is-focus` judgment ignores `el-cascader__search-input`.

File: packages/components/cascader/src/cascader.vue [hunk 1]
```diff
@@ -40,7 +40,8 @@
           :disabled="isDisabled"
           :validate-event="false"
           :size="realSize"
-          :class="nsCascader.is('focus', popperVisible)"
+          :class="inputClass"
+          :tabindex="filterable ? '-1' : '0'"
```
Comment (tolking) [inline packages/components/cascader/src/cascader.vue:44]:
```suggestion
          :tabindex="multiple && filterable && !isDisabled ? -1 : undefined"
```

It should be ensured that only one input can be focused

**Review by tolking — APPROVED** [review]
LGTM.

**Pull request merged**